src/middlewares/methods/block_tag.rs

```diff
@@ -68,7 +70,23 @@ impl BlockTagMiddleware {
                         context.insert(BypassCache(true));
                         None
                     }
-                    _ => None,
+                    number => {
+                        // bypass cache for block number to avoid caching forks unless it's a finalized block
+                        let mut bypass_cache = true;
+                        if let Some((_, finalized_number)) = self.api.current_finalized_head() {
+                            if let Some(hex_number) = number.strip_prefix("0x") {
+                                if let Ok(number) = u64::from_str_radix(hex_number, 16) {
+                                    if number <= finalized_number {
+                                        bypass_cache = false;
```

Comment:
```suggestion
                                        context.insert(BypassCache(true));
```

no need `bypass_cache`?

Reply:
It needs to be the opposite and I want to insert `BypassCache` only if `bypass_cache` is true to avoid overriding any existing `BypassCache`